### DISCUSSION TITLE
Templated Page - Sidebar

### DISCUSSION THREAD

**Discussion by CodingPupper3033** [inventree/InvenTree#5802] [Q&A] [answered]
Hello again,

I am trying to create a new "app" (plugin with a navbar item and app page) using the plugin system. I want to make a new tab using Navigation Mixin (which I have been able to do), which brings you to a new page similar to <URL>/stock/# (a site with the navbar, sidebar, content, etc). I could implement this into an existing app (stock, part, etc) using panel mixins. However, I plan on implementing enough separate tabs to need a separate page.

I looked into how "apps" (like parts and stock) implement this, and it appears pages extend `page_base.html`. Then, to add content to blocks like `page_content`, you declare a block and put templated HTML into it. It seems to add a sidebar you would add content to `block sidebar`; however, my attempts seem dysfunctional. Specifically, the sidebar will not appear; however, other blocks (like page content) will appear.

I have posted relevant code below. I included "my own" sidebar items and attempted to copy the contents of a stock sidebar in an attempt to get anything to appear. I have tried making the sidebar its own file, only including the stock one, only have "my sidebar items" (I copied and pasted yours for testing).

**I am currently unsure why the sidebar will not appear. Would you be able to tell if you see an obvious issue or if there may be a backend issue causing the sidebar not to appear?** As always, I am grateful for any time and help you may be able to provide!

My Attempt:
``` HTML
{% extends "page_base.html" %}
{% load static %}
{% load inventree_extras %}
{% load plugin_extras %}
{% load i18n %}


{% block page_title %}
{% inventree_title %} | {% trans "Loaning" %} - {{ item }}
{% endblock page_title %}


{% block sidebar %}
{% trans "Sublocations" as text %}
{% include "sidebar_item.html" with label='sublocations' text=text icon="fa-sitemap" %}
{% trans "Stock Items" as text %}
{% include "sidebar_item.html" with label='stock' text=text icon="fa-boxes" %}

{% include "stock/location_sidebar.html" %}
{% endblock sidebar %}
```

**Comment by CodingPupper3033** [comment, marked as answer]
At the bottom of the page, you must enable the sidebar. I can't believe I missed this one, sorry about that

``` HTML
{% block js_ready %}
{{ block.super }}
    enableSidebar('stocklocation');


{% endblock js_ready %}

```

**Comment by SchrodingersGat** [comment]
@CodingPupper3033 glad you worked it out! Do you think you could submit an update to the docs so that other users can see this too?

**Reply by CodingPupper3033** [reply]
I'll try. I'm trying to setup a vm to run the docs server (so I can add to the docs and see how it actually gets formatted), and it isn't exactly working yet. The `mkdocs serve -f docs/mkdocs.yml -a localhost:8080` command will run, but I don't think it's actually opening the port. If I get this to work, sure!

**Reply by CodingPupper3033** [reply]
It is just markdown files; I feel stupid, I am working on it now.

Do you have any preference on what I call the new pages?

**Reply by CodingPupper3033** [reply]
I have created some documentation on this process. How would I suggest these changes with git? May I just `git commit -m "..."`, or is there another way to request changes?

(I really don't want to accidentally mess up the repository as I am only mediocre at git)

**Reply by CodingPupper3033** [reply]
Hello again,

Small update: I looked over the contributing guide, and it appears to contribute, you should make an issue from the discussion, and a branch from the issue. I have tried to do that, and push my branch, but I may be doing something wrong.

My attempts at pushing:
![image](https://github.com/inventree/InvenTree/assets/37462274/e806efaf-59df-4798-a584-310742f10e6d)


Otherwise, here is a patch file (that I hope works and you can use); it is up to you how It is committed. I don't need credit if this is just easier.
[0001-Added-Views-blocks-and-templates-documentation.patch](https://github.com/inventree/InvenTree/files/13194440/0001-Added-Views-blocks-and-templates-documentation.patch)
[patch.patch](https://github.com/inventree/InvenTree/files/13194441/patch.patch)

Also, if you want anything changed, let me know, I don't know how detailed you wanted me to be, or on what parts specifically

**Reply by matmair** [reply]
This is more a git/permission error. You are trying to push to InvenTree, which even we do not do. You need to fork InvenTree (there is a button for that in Github on the upper right when you view the repo) and push to that. You then open a `Pull Request` in github from your new branch to the main inventree repo. We run some checks automatically and then it can be merged. This is the default flow for git as it would be pure chaos if everyone (including maintainers) was just pushing to the main repo.

You can also just go to the file you want to edit on github and hit `edit` in the upper right corner (above the content of the file) - this will let you submit a path directly.

Let me know if you want to try one of those methods or if I should just propose your patch and co-author you on it.

**Reply by CodingPupper3033** [reply]
Ah, I completely forgot about forking. Let me try that first; then, if I can't figure it out, you may submit it!

**Reply by CodingPupper3033** [reply]
@matmair I believe I submitted a pull request. I am unsure how I link the discussion and issue to the pull request

**Reply by SchrodingersGat** [reply]
@CodingPupper3033 in the description for the PR, simple put "Closes #issue" (I have done this for you in your PR now, so you can see what to do next time! )

**Reply by CodingPupper3033** [reply]
Thank you :)

Is the pull request all set? Do I need to do anything more? Does the documentation itself look good?
